extensions/resteasy-reactive/rest-client-microprofile/runtime/src/main/java/io/quarkus/resteasy/reactive/client/microprofile/RestClientCDIDelegateBuilder.java

```diff
@@ -0,0 +1,279 @@
+    public static final String REST_KEY_STORE_TYPE = "%s/" + MP_REST + "/keyStoreType";
+    public static final String REST_HOSTNAME_VERIFIER = "%s/" + MP_REST + "/hostnameVerifier";
+    public static final String REST_NOOP_HOSTNAME_VERIFIER = "io.quarkus.restclient.NoopHostnameVerifier";
+    public static final String TLS_TRUST_ALL = "quarkus.tls.trust-all";
+
+    private final Class<?> jaxrsInterface;
+    private final String baseUriFromAnnotation;
+    private final String propertyPrefix;
+
+    // used by generated code
+    @SuppressWarnings("unused")
+    public static Object createDelegate(Class<?> jaxrsInterface, String baseUriFromAnnotation, String propertyPrefix) {
+        return new RestClientCDIDelegateBuilder(jaxrsInterface, baseUriFromAnnotation, propertyPrefix).build();
+    }
+
+    private RestClientCDIDelegateBuilder(Class<?> jaxrsInterface, String baseUriFromAnnotation, String propertyPrefix) {
+        this.jaxrsInterface = jaxrsInterface;
+        this.baseUriFromAnnotation = baseUriFromAnnotation;
+        this.propertyPrefix = propertyPrefix;
+    }
+
+    private Object build() {
+        RestClientBuilder builder = RestClientBuilder.newBuilder();
+        configureBaseUrl(builder);
+        configureTimeouts(builder);
+        configureProviders(builder);
+        configureSsl(builder);
+        // If we have context propagation, then propagate context to the async client threads
+        InstanceHandle<ManagedExecutor> managedExecutor = Arc.container().instance(ManagedExecutor.class);
+        if (managedExecutor.isAvailable()) {
+            builder.executorService(managedExecutor.get());
+        }
+
+        Object result = builder.build(jaxrsInterface);
+        return result;
+    }
+
+    private void configureSsl(RestClientBuilder builder) {
+
+        Optional<Boolean> trustAll = getOptionalProperty(TLS_TRUST_ALL, Boolean.class);
+        if (trustAll.isPresent() && trustAll.get()) {
+            registerHostnameVerifier(REST_NOOP_HOSTNAME_VERIFIER, builder);
+        }
+
+        Optional<String> maybeTrustStore = getOptionalDynamicProperty(REST_TRUST_STORE, String.class);
+        if (maybeTrustStore.isPresent()) {
+            registerTrustStore(maybeTrustStore.get(), builder);
+        }
+
+        Optional<String> maybeKeyStore = getOptionalDynamicProperty(REST_KEY_STORE, String.class);
+        if (maybeKeyStore.isPresent()) {
+            registerKeyStore(maybeKeyStore.get(), builder);
+        }
+
+        Optional<String> maybeHostnameVerifier = getOptionalDynamicProperty(REST_HOSTNAME_VERIFIER, String.class);
+        if (maybeHostnameVerifier.isPresent()) {
+            registerHostnameVerifier(maybeHostnameVerifier.get(), builder);
+        }
+
+        // we need to push a disabled SSL context when SSL has been disabled
+        // because otherwise Apache HTTP Client will try to initialize one and will fail
```

Comment:
I don't think this is valid for this impl. Even if it is, the comment sure isn't :)